File: SDEIArchitecture/app/src/main/java/com/sdei/sdeiarchitecture/repository/networkOperator/NetworkAdapter.java
package com.sdei.sdeiarchitecture.repository.networkOperator;


import com.sdei.sdeiarchitecture.BuildConfig;
import com.sdei.sdeiarchitecture.helper.dagger.scope.AppScope;

import java.io.IOException;
import java.util.concurrent.TimeUnit;

import javax.inject.Scope;

import okhttp3.Interceptor;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;


@AppScope
public class NetworkAdapter {

    public static NetworkAdapter cInstance;
    public ApiService apiService;

    public String demo() {
        return "HEllossssssssssssss";
    }

    /* Static 'instance' method */
    public static NetworkAdapter getInstance() {
        if (cInstance == null) {
            cInstance = new NetworkAdapter();
            cInstance.setupRetrofitClient();
            return cInstance;
        }
        cInstance.setupRetrofitClient();
        return cInstance;
    }

    public static void initInstance() {
        if (cInstance == null) {
            cInstance = new NetworkAdapter();
            cInstance.setupRetrofitClient();
        }
    }

    public void setupRetrofitClient() {

        try {
            Interceptor HEADER_INTERCEPTOR = new Interceptor() {
                @Override
                public Response intercept(Chain chain) throws IOException {

                    Request original = chain.request();

                    Request.Builder requestBuilder = original.newBuilder()
                            .header("Content-Type", "application/json");
                    requestBuilder.method(original.method(), original.body());

                    Request request = requestBuilder.build();
                    return chain.proceed(request);
                }
            };


            HttpLoggingInterceptor interceptor = new HttpLoggingInterceptor();
            interceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
            OkHttpClient client = new OkHttpClient.Builder().addInterceptor(interceptor).addInterceptor(HEADER_INTERCEPTOR)
                    .cache(null)
                    .connectTimeout(60, TimeUnit.MINUTES)
                    .readTimeout(60, TimeUnit.MINUTES).build();


            Retrofit retrofit = new Retrofit.Builder()
                    .baseUrl(BuildConfig.BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .client(client)
                    .build();
            apiService = retrofit.create(ApiService.class);
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }

    public ApiService getNetworkServices() {
        return apiService;
    }
}

File: SDEIArchitecture/app/src/main/java/com/sdei/sdeiarchitecture/repository/networkoperator/NetworkAdapter.kt
package com.sdei.sdeiarchitecture.repository.networkoperator


import com.sdei.sdeiarchitecture.BuildConfig
import okhttp3.Interceptor
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import java.util.concurrent.TimeUnit


class NetworkAdapter {
    private val NETWORK_URL = ""

    companion object {
        var cInstance: NetworkAdapter? = null
        var apiService: ApiService? = null

        /* Static 'instance' method */
        fun getInstance(): NetworkAdapter {
            if (cInstance == null) {
                cInstance =
                    NetworkAdapter()
                cInstance?.setupRetrofitClient()
                return cInstance!!
            }
            cInstance?.setupRetrofitClient()
            return cInstance!!
        }

    }

    fun setupRetrofitClient() {
        try {
            val HEADER_INTERCEPTOR = Interceptor { chain ->
                val original = chain.request()

                val requestBuilder = original.newBuilder()
                    .header("Content-Type", "application/json")
                requestBuilder.header("Accept", "application/json")
                requestBuilder.method(original.method(), original.body())

                val request = requestBuilder.build()
                chain.proceed(request)
            }


            val interceptor = HttpLoggingInterceptor()
            interceptor.level = HttpLoggingInterceptor.Level.BODY
            val client = OkHttpClient.Builder().addInterceptor(interceptor)
                .addInterceptor(HEADER_INTERCEPTOR)
                .cache(null)
                .connectTimeout(60, TimeUnit.MINUTES)
                .readTimeout(60, TimeUnit.MINUTES).build()


            val retrofit = Retrofit.Builder()
                .baseUrl(BuildConfig.BASE_URL)
                .addConverterFactory(GsonConverterFactory.create())
                .client(client)
                .build()
            apiService = retrofit.create(ApiService::class.java)
        } catch (e: Exception) {
            throw RuntimeException(e)
        }

    }

    fun getNetworkServices(): ApiService? {
        return apiService
    }
}